src/coreclr/vm/invokeutil.cpp

```diff
@@ -167,45 +158,35 @@ void InvokeUtil::CopyArg(TypeHandle th, OBJECTREF *pObjUNSAFE, ArgDestination *a
     IN_TARGET_32BIT(case ELEMENT_TYPE_I:)
     IN_TARGET_32BIT(case ELEMENT_TYPE_U:)
     {
-        // If we got the univeral zero...Then assign it and exit.
-        if (rObj == 0)
-            *(PVOID *)pArgDst = 0;
-        else
         {
+            _ASSERTE(pArgUnsafe != NULL);
             ARG_SLOT slot;
-            CreatePrimitiveValue(type, oType, rObj, &slot);
+            CreatePrimitiveValue(type, oType, pArgUnsafe, pMT, &slot);
```

Comment:
I think this can be streamlined. We know that the types match exactly at this point, so we can just assign the value to `pArgUnsafe` in the case statements above. We should not need to go through `CreatePrimitiveValue` that does the same switch again and also tries to cover some type mismatches.

Reply:
> I think this can be streamlined

Yes the current code now avoids this. Thanks